Combobox.js

```diff
@@ -848,6 +873,21 @@ define([
 			this.openDropDown();
 		},
 
+		_retrieveItemFromSource: function (key) {
+			var item = null,
+				_source = this.source || (this.list && this.list.source);
+			if (_source) {
+				if (Array.isArray(_source)) {
+					item = _source.filter(function (i) {
+						return i[this.list.valueAttr || this.list.labelAttr] === key;
+					}.bind(this))[0];
+				} else {
+					item = _source.getSync && _source.getSync(key);
+				}
+			}
+			return item;
+		},
```

Comment:
I think I can make this part specific for only when in not auto-complete mode (hasDownArrow=false). Do you agree?